### PR TITLE
Fix claimed holderof drop

### LINKED ISSUE #9404: Claimed check failed on Holder of drop
After digging into the code, I discovered that the existing claim check no longer worked. 

It would cause several bugs:
1. incorrect available minted amount on holder of drop page. (related to https://github.com/kodadot/nft-gallery/issues/9390)
2. minting error when try to minting multiple holder of drop. (related to https://github.com/kodadot/private-workers/issues/64)

Here is method to claimed check.
image

It always returns null regardless of which NFT is checked.
image

image

## expect solution
Find the correct method to check if the NFT has been claimed on the holderOf drops page.

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #9404
  - [x] Closes https://github.com/kodadot/private-workers/issues/64
  - [x] related with #9390

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

image
image

  ## Copilot Summary
  copilot:summary

  copilot:poem

### CODE REVIEW

File: utils/drop.ts [hunk 1]
```diff
@@ -6,10 +6,12 @@ export const DEFAULT_DROP = {
 
 export const HOLDER_OF_DROP_MAP = {
   '50': '51',
+  '352': '377',
```
Comment (preschian) [inline utils/drop.ts:9]: TIL, we have this. is there a way to make it dynamic?

File: components/drops/useDrops.ts [hunk 2]
```diff
@@ -206,17 +206,41 @@ export const useDropMinimumFunds = (drop) => {
   }
 }
 
+// Convert the collection to hex
+// Construct the number - if is length 2 - pad it right with two zeros
+// If the length is 3 - pad it right with one zero and put the first digit in the end
+// If the lenght is 4 - swap the pair - aka first two letters go last
+// When you have number from step two construct the full format - 0x00${theMagic}0000
+const convertCollectionIdToMagicId = (id: string) => {
```
Comment (preschian) [inline components/drops/useDrops.ts:214]: Maybe Polkadot-js has a function to convert to hex?

**Review by preschian — COMMENTED** [review]
nice 👍🏻

File: utils/drop.ts (hunk 1 above)
Comment (preschian) [reply utils/drop.ts:9]: oohh, this is for gallery item. lgtm

**Review by preschian — APPROVED** [review]

File: utils/drop.ts (hunk 1 above)
Comment (Jarsen136) [reply utils/drop.ts:9]: > oohh, this is for gallery item.
Yes

File: components/drops/useDrops.ts (hunk 2 above)
Comment (Jarsen136) [reply components/drops/useDrops.ts:214]: I'm not sure if there is a function to do this magic convert based on the comment.

**Comment by yangwao** [comment]
cc @prury can you review please?

File: components/drops/useDrops.ts (hunk 2 above)
Comment (vikiival) [reply components/drops/useDrops.ts:214]: Yes use please utils from polkdadot utils if possible

**Comment by Jarsen136** [comment]
Oh there is no preview url because it's intend to merge to `kodadot:fix-remaining-nfts` branch.

Let's merge it because I have tested on the localhost.

**Comment by vikiival** [comment]
@Jarsen136 i am going to merge this please open an PR for the base branch

**Comment by Jarsen136** [comment]
> @Jarsen136 i am going to merge this please open an PR for the base branch

https://github.com/kodadot/nft-gallery/pull/9397

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 7.85 USD/DOT ~ 3.822 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0xb870d2b33a0583b34b774ccee1e73d158b282cc269914afb617da509e08d30dc](https://polkadot.subscan.io/extrinsic/0xb870d2b33a0583b34b774ccee1e73d158b282cc269914afb617da509e08d30dc)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**